src/main/java/org/jabref/logic/util/io/FileNameHandler.java

```diff
@@ -67,7 +71,41 @@ public static String cleanDirectoryName(String badFileName) {
         return cleanName.toString().trim();
     }
 
+    /**
+     * Returns an absolute path to a file which does not match with any existing file names
+     *
+     * @param targetDirectory The directory in which file name should be unique
+     * @param fileName Suggested name for the file
+     * @return a file-name such that it does not match any existing files in targetDirectory.
+     * */
+    public static String getNonOverWritingFileName(Path targetDirectory, String fileName) {
+        String absoluteName = targetDirectory.resolve(fileName)
+                                            .toString();
+
+        Optional<String> extensionOptional = FileUtil.getFileExtension(fileName);
+
+        // the suffix include the '.' , if extension is present Eg: ".pdf"
+        String extensionSuffix = "";
+        String fileNameWithoutExtension = fileName;
+
+        if (extensionOptional.isPresent()) {
+            extensionSuffix = '.' + extensionOptional.get();
+            fileNameWithoutExtension = fileName.substring(0, fileName.lastIndexOf('.'));
+        }
+
+        Path absolutePath = Paths.get(absoluteName);
+        String newFileName = fileName;
+
+        for (int counter = 1; Files.exists(absolutePath); ++counter) {
```

Comment:
```suggestion
        for (int counter = 1; Files.exists(absolutePath); counter++) {
```

`++` at the end to be inline with the usual `i++`.

Reply:
After thinking longer, please rewrite to a `while` loop. Not testing for the counter in the test of a `for` loop is uncommon. Thus, hard to read when maintaining the code.